src/Exception/PotentialAttackException/InvalidHashException.php

```diff
@@ -2,11 +2,62 @@
 
 namespace Tuf\Exception\PotentialAttackException;
 
+use Psr\Http\Message\StreamInterface;
 use Tuf\Exception\TufException;
 
 /**
  * Indicates an invalid hash was computed for a downloaded target.
+ *
+ * This exception provides access to the underlying stream which caused the
+ * error. The stream's contents are NOT trusted by TUF, so any code which
+ * interacts with this exception, or the stream, should do so with great caution
+ * and treat the stream contents as unsafe.
  */
 class InvalidHashException extends TufException
 {
+    /**
+     * An untrusted stream object pointing to the downloaded target.
+     *
+     * WARNING: The contents of the stream are NOT trusted by TUF! Any code
```

Comment:
It is not only that it is not trusted. It actually "failed TUF validation". maybe we should say that.